ui/src/app/views/run/workflow/variables/job.variables.component.ts

```diff
@@ -0,0 +1,70 @@
+import {Component, Input, ViewChild} from '@angular/core';
+import {Variable} from '../../../../model/variable.model';
+import {Parameter} from '../../../../model/parameter.model';
+import {SemanticModalComponent} from 'ng-semantic/ng-semantic';
+
+@Component({
+    selector: 'app-job-variable',
+    templateUrl: './job.variable.html',
+    styleUrls: ['./job.variable.scss']
+})
+export class JobVariableComponent {
+
+    @ViewChild('jobVariablesModal')
+    jobVariablesModal: SemanticModalComponent;
+
+    @Input('variables')
+    set variables( data: Array<Variable>) {
+        this.init();
+        if (data) {
```

Comment:
even if we use typescript?  just want to know if it's null or undefined, not if it's an array